### ISSUE TITLE
Found with fuzzer: Can't place branch!

### ISSUE THREAD

**Issue by elunna** [elunna/hackem#63]
```
Can't place branch!
Generating more information you may report:

[0] /lib/x86_64-linux-gnu/libasan.so.5(+0x6cd40) [0x7ffff75f8d40]
[1] /home/lunatunez/games/hackemdir/hackem(+0x4fbb3a) [0x555555a4fb3a]
[2] /home/lunatunez/games/hackemdir/hackem(+0x4fba2d) [0x555555a4fa2d]
[3] /home/lunatunez/games/hackemdir/hackem(panic+0x6ab) [0x555555a5684f]
[4] /home/lunatunez/games/hackemdir/hackem(impossible+0x2c8) [0x555555faeca1]
[5] /home/lunatunez/games/hackemdir/hackem(+0x7e444c) [0x555555d3844c]
[6] /home/lunatunez/games/hackemdir/hackem(place_branch+0xd8) [0x555555d385cc]
[7] /home/lunatunez/games/hackemdir/hackem(+0x7e1662) [0x555555d35662]
[8] /home/lunatunez/games/hackemdir/hackem(mklev+0x86) [0x555555d374ca]
[9] /home/lunatunez/games/hackemdir/hackem(+0x2e4538) [0x555555838538]
[10] /home/lunatunez/games/hackemdir/hackem(rhack+0x1b5f) [0x55555586a1c5]
[11] /home/lunatunez/games/hackemdir/hackem(moveloop+0x7355) [0x555555768f7b]
[12] /home/lunatunez/games/hackemdir/hackem(main+0xdda) [0x5555563dd234]
[13] /lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xf3) [0x7ffff7365083]
[14] /home/lunatunez/games/hackemdir/hackem(_start+0x2e) [0x55555575face]


Program received signal SIGABRT, Aborted.
__GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:50
50	../sysdeps/unix/sysv/linux/raise.c: No such file or directory.
#0  __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:50
#1  0x00007ffff7363859 in __GI_abort () at abort.c:79
#2  0x0000555555a4fa53 in NH_abort () at end.c:236
#3  0x0000555555a5684f in panic (str=0x555556547cc0 "%s") at end.c:802
#4  0x0000555555faeca1 in impossible (s=0x5555564f8980 "Can't place branch!") at pline.c:518
#5  0x0000555555d3844c in find_branch_room (mp=0x7fffffffe010) at mklev.c:1377
#6  0x0000555555d385cc in place_branch (br=0x603000001300, x=0 '\000', y=0 '\000')
    at mklev.c:1419
#7  0x0000555555d35662 in makelevel () at mklev.c:1013
#8  0x0000555555d374ca in mklev () at mklev.c:1255
#9  0x0000555555838538 in wiz_makemap () at cmd.c:899
#10 0x000055555586a1c5 in rhack (cmd=0x5555567cf540 <in_line> "v") at cmd.c:5537
#11 0x0000555555768f7b in moveloop (resuming=0 '\000') at allmain.c:823
#12 0x00005555563dd234 in main (argc=0, argv=0x7fffffffe628) at ../sys/unix/unixmain.c:353
#0  __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:50
        set = {__val = {0, 0, 0, 0, 0, 140737335783280, 93824997455653, 335544320, 
            140737341046928, 1073741824, 0, 140737342793376, 4294967295, 140737337360384, 
            140737342776480, 3350589553395594496}}
        pid = <optimized out>
        tid = <optimized out>
        ret = <optimized out>
#1  0x00007ffff7363859 in __GI_abort () at abort.c:79
        save_stage = 1
        act = {__sigaction_handler = {sa_handler = 0xffffffff, sa_sigaction = 0xffffffff}, 
          sa_mask = {__val = {140737337360384, 140737342776480, 3350589553395594496, 
              140737488344192, 140737342793376, 18446744073709547520, 140737327104000, 
              140737345462488, 140737488345264, 140737488345264, 140737344331794, 
              17592186043024, 140737488338944, 140737341046928, 3350589553395594496, 
              140737488344592}}, sa_flags = 1447523584, sa_restorer = 0x7fffffffd7d0}
        sigs = {__val = {32, 0 <repeats 15 times>}}
#2  0x0000555555a4fa53 in NH_abort () at end.c:236
        gdb_prio = 1
        libc_prio = 2
        aborting = 1 '\001'
#3  0x0000555555a5684f in panic (str=0x555556547cc0 "%s") at end.c:802
        the_args = {{gp_offset = 16, fp_offset = 48, overflow_arg_area = 0x7fffffffd8a0, 
            reg_save_area = 0x7fffffffd7d0}}
#4  0x0000555555faeca1 in impossible (s=0x5555564f8980 "Can't place branch!") at pline.c:518
        pbuf = "Can't place branch!\000UU", '\000' <repeats 30 times>, "\001\000\000\000\001\000\000\000\001\000\000\000\000\000\000\000\001\000\000\000\001\000\000\000\001\000\000\000\001\000\000\000\000\000\000\000\227`YVUU\000\000\260\331\377\377\377\177\000\000\000uGVr\257\177.\240\331\377\377\377\177\000\000`\336\377\377\377\177\000\000\220\373\377\377\377\017\000\000\200\334\377\377\377\177\000\000\340\331\377\377\377\177\000\000\000uGVr\257\177.\340\331\377\377\377\177\000\000`\336\377\377\377\177\000\000\000\332\377\377\377\177\000\000\000uGVr\257\177.\320\336\377\377\377\177\000\000"...
        the_args = {{gp_offset = 8, fp_offset = 48, overflow_arg_area = 0x7fffffffdf90, 
            reg_save_area = 0x7fffffffdeb0}}
#5  0x0000555555d3844c in find_branch_room (mp=0x7fffffffe010) at mklev.c:1377
        croom = 0x5555567d2b70 <rooms+1296>
#6  0x0000555555d385cc in place_branch (br=0x603000001300, x=0 '\000', y=0 '\000')
    at mklev.c:1419
        m = {x = 48 '0', y = 4 '\004'}
        dest = 0x7fffffffdff0
        make_stairs = 0 '\000'
        br_room = 0x5555560aee92 <rn2+47>
#7  0x0000555555d35662 in makelevel () at mklev.c:1013
        croom = 0xb
        troom = <optimized out>
        tryct = 0
        i = 0
        tmonst = 0x5555564628a0
        branchp = 0x603000001300
        room_threshold = 4
        boxtype = 32767
        pos = {x = 76 'L', y = 15 '\017'}
#8  0x0000555555d374ca in mklev () at mklev.c:1255
        croom = 0xffffffffc6c
        ridx = 32767
#9  0x0000555555838538 in wiz_makemap () at cmd.c:899
        mtmp = 0x0
        was_in_W_tower = 0 '\000'
#10 0x000055555586a1c5 in rhack (cmd=0x5555567cf540 <in_line> "v") at cmd.c:5537
        tlist = 0x55555668ca20 <extcmdlist+5184>
        res = 688
        func = 0x555555838059 <wiz_makemap>
        spkey = 0
        prefix_seen = 0 '\000'
        bad_command = 114 'r'
        firsttime = 1 '\001'
#11 0x0000555555768f7b in moveloop (resuming=0 '\000') at allmain.c:823
        moveamt = 10
        wtcap = 0
        change = 0
        monscanmove = 0 '\000'
        timeout_start = 31844
        past_clock = 337034
        elf_regen = 1 '\001'
        orc_regen = 1 '\001'
        vamp_regen = 1 '\001'
#12 0x00005555563dd234 in main (argc=0, argv=0x7fffffffe628) at ../sys/unix/unixmain.c:353
        fd = -1
        dir = 0x0
        exact_username = 0 '\000'
        resuming = 0 '\000'
        plsel_once = 1 '\001'

```

**Comment by elunna** [comment]
I was able to get this info will still in gdb:
branch->id = 9 (Probably the chaotic quest)
branch->type = 3
#define BR_PORTAL 3  /* Connection by magic portals (traps) */


It's likely this problem is caused by ludius being forced into the first vault to appear after level 10. Since there are 4 portal levels from 15-19 (3 alignment quests + regular quest), if the ludious portal is put into one of those levels, it's likely that one of the other portals will not be generated...

**Comment by elunna** [comment]
Possible fix in this commit  00d98cb94 - Made some adjustments to the themed rooms. Moved some of the longer and bigger rooms to minlevel 20 - to prevent issues generating portals in the quest levels. Also converted more of the generic type rooms to ordinary. Hopefully this allows more "ordinary" rooms to be present in levels 15-19.

**Comment by elunna** [comment]
Here is proof that the themed rooms can only generate 2 rooms - actually ran into this while manually testing something!
<img width="724" alt="2room level" src="https://user-images.githubusercontent.com/5000096/201548072-d772a9ab-a7d7-44b9-9ce4-272a374169cc.png">
